src/rdymesh.c

```diff
@@ -223,7 +227,7 @@ PetscErrorCode RDyVerticesCreate(PetscInt num_vertices, RDyVertices *vertices) {
 /// @param [out] vertices A pointer to an RDyVertices that stores allocated data.
 ///
 /// @return 0 on success, or a non-zero error code on failure
-PetscErrorCode RDyVerticesCreateFromDM(DM dm, RDyVertices *vertices) {
+PetscErrorCode RDyVerticesCreateFromDM(DM dm, RDyVertices *vertices, PetscInt *num_vertices_total, PetscBool mesh_refined) {
```

Comment:
You know, instead of carrying around flags that indicate whether a DM is refined or not, we could use [DMSetRefineLevel](https://petsc.org/release/manualpages/DM/DMSetRefineLevel/) and [DMGetRefineLevel](https://petsc.org/release/manualpages/DM/DMGetRefineLevel/) to store the number of refinements within the DM itself.